stake-pool/py/stake_pool/instructions.py

```diff
@@ -496,6 +542,12 @@ class InstructionType(IntEnum):
     "seed" / Int32ul
 )
 
+TOKEN_METADATA_LAYOUT = Struct(
+    "name" / Bytes(32),
+    "symbol" / Bytes(32),
+    "uri" / Bytes(32)
+)
```

Comment:
You'll need to be careful with these. They're encoded as `String` in the interface, which in Borsh means they're actually like vectors, meaning 4 bytes to encode the length, followed by the utf8-encoded bytes.  You can find more info at the borsh spec: https://borsh.io/ -> "Specification"

Reply:
Thanks, I was actually in doubts about that so I put `Bytes` as a "placeholder" to further updates. Thanks for the clarification.

Reply:
@joncinque I gave a look at borsh specification, does this implementation match for `String` type?
```
TOKEN_METADATA_LAYOUT = Struct(
    "name" / Prefixed(Int32ul, GreedyString("utf8")),
    "symbol" / Prefixed(Int32ul, GreedyString("utf8")),
    "uri" / Prefixed(Int32ul, GreedyString("utf8"))
)
```
Here you can find the doc of `contruct` module for [Prefixed](https://construct.readthedocs.io/en/latest/api/core.html#construct.core.Pre) and [GreedyString](https://construct.readthedocs.io/en/latest/api/core.html#construct.core.GreedyString).
`Prefixed` contains a field describing the length (so it parses this filed and then reads this amount of bytes), while `GreedyString` is essentially a `GreedyBytes` that encodes bytes with unicode byte encoding.

Reply:
Yep, that looks perfect!